features/admin.users.v1/components/user-profile.tsx

```diff
@@ -1511,14 +1662,99 @@ export const UserProfile: FunctionComponent<UserProfilePropsInterface> = (
+
+                if (authenticatedUserRole.display === accountAppImpersonateRoleName) {
+                    return true;
+                }
+            }
+
+            return false;
+        };
+
+        const isMyAccountImpersonatable = (): boolean => {
+
+            // Load My Account Application and check
+            // 1. If it has skip login consent.
+            // 2. If it has disabled application.
+            // 3. If it has shared with the sub org.
+            return !isUserCurrentLoggedInUser && isLoggedInUserAuthorizedToImpersonate();
+        };
+
+        return (
+            (
+                !isReadOnly
+                || isUserManagedByParentOrg
+                || user[ SCIMConfigs.scim.systemSchema ]?.userSourceId
+            ) && (
+                !isCurrentUserAdmin
+                || !isUserCurrentLoggedInUser
+            ) && isLoggedInUserAuthorizedToImpersonate() && isFeatureEnabled(userFeatureConfig, "IMPERSONATE_USER") ?
+                (
+                    <React.Fragment>
+                        <DangerZoneGroup
+                            className="action-zone"
+                        >
+                            {
+                                !isUserManagedByParentOrg && (
+                                    <DangerZone
+                                        data-testid={ `${ testId }-danger-zone-toggle` }
+                                        className="action-zone"
+                                        actionTitle={ t("user:editUser." +
+                                    "userActionZoneGroup.impersonateUserZone.actionTitle") }
+                                        header={
+                                            t("user:editUser.userActionZoneGroup.impersonateUserZone.header")
+                                        }
+                                        subheader={
+                                            t("user:editUser.userActionZoneGroup.impersonateUserZone.subheader")
+                                        }
+                                        onActionClick={ (): void => {
+                                            handleUserImpersonation();
+                                        } }
+                                        isButtonDisabled={ !isMyAccountImpersonatable() }
+                                        isButtonLoading={ impersonationInProgress }
+                                    />
+                                )
+                            }
+                        </DangerZoneGroup>
+                        {
+                            impersonationInProgress && codeChallenge != undefined && codeChallenge != null && (
+                                <div>
+                                    <iframe
+                                        hidden
+                                        src={ "https://localhost:9001/console/resources/init-impersonate.html"
```

Comment:
Hardcoded console url wouldn't work in other deployments

Reply:
82d4ff507d55ec7ed6c0b57080b6e89963594857